### DISCUSSION TITLE
Diagrams

### DISCUSSION THREAD

**Discussion by becarpenter** [becarpenter/book6#7] [General]
I just experimented with Mermaid, the simple drawing tool that GitHub supports in markdown files. It's limited in what it can do (flowcharts and similar) but quite easy.

See the result at https://github.com/becarpenter/book6/blob/main/8.%20Further%20Reading/8.%20Further%20Reading.md

Documentation at https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams

Is this a good solution for us?

**Comment by becarpenter** [comment]
You can use https://mermaid.live/ to convert mermaid diagrams to SVG or PNG, so I think it's a robust solution for simple diagrams.

**Comment by becarpenter** [comment]
I updated CONTRIBUTING.md to indicate a choice of ASCII art, mermaid, or SVG for diagrams.

**Reply by jenslink** [reply]
There are tools to convert ASCII art into SVG. Previously I used https://github.com/stathissideris/ditaa to convert some diagrams from RFC to SVG, but https://github.com/blampe/goat also looks good (and it's not written in Java).

**Comment by becarpenter** [comment]
I moved the information into a section in Chapter 99

**Comment by Paolo-Volpato** [comment]
Hi, I have just added some text to the "Status" section of Chapter 5 (Deployment).
To make it quickly, I have also uploaded a jpg image.
Let me know if it is fine or it is better to use SVG/PNG.
Thanks!

**Reply by becarpenter** [reply]
No, I see no problem with using JPG. I didn't test it before, which is why I documented PNG, but I will add it to the suggested methods. Thanks!

**Comment by becarpenter** [comment]
For now having mermaid, SVG, PNG and JPG seems to cover everything, so I will close the discussion topic.
